### PR TITLE
feature - show venue list into collectionview

### PR DESCRIPTION
**Pull request by at-phutran**
- show venue list into collectionview

### CODE REVIEW

File: ATFoody/Model/Schema/Trending.swift [hunk 1]
```diff
@@ -49,3 +52,8 @@ extension Trending {
         return [Photo.self]
     }
 }
+extension Trending {
+    var ratingColorBackground: UIColor {
```
Comment (at-quangphu) [inline ATFoody/Model/Schema/Trending.swift]: `Model` don't know about ui. Please make a color at controller

File: ATFoody/View/Controller/Home/CollectionViewCell/CollectionViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+//  CollectionViewCell.swift
+//  ATFoody
+//
+//  Created by Phu Tran on 11/10/17.
+//  Copyright © 2017 Asian Tech. All rights reserved.
+//
+
+import UIKit
+
+class CollectionViewCell: UICollectionViewCell {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/CollectionViewCell/CollectionViewCell.swift]: Missing `final` keyword

File: ATFoody/View/Controller/Home/CollectionViewCell/CollectionViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    @IBOutlet weak var thumbnailImageView: UIImageView!
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var addressLabel: UILabel!
+    @IBOutlet weak var ratingLabel: UILabel!
+
+    override func awakeFromNib() {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/CollectionViewCell/CollectionViewCell.swift]: Remove this function if no using

File: ATFoody/View/Controller/Home/CollectionViewCell/CollectionViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+//  Created by Phu Tran on 11/10/17.
+//  Copyright © 2017 Asian Tech. All rights reserved.
+//
+
+import Foundation
+import SwiftUtils
+import UIKit
+import MVVM
+
+final class CollectionViewModel: ViewModel {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/CollectionViewCell/CollectionViewModel.swift]: Do you rename this class. Its name is regular.

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 5]
```diff
@@ -13,6 +13,10 @@ import RealmS
 final class LocationViewController: UIViewController {
     // MARK: - IBOutlets
     @IBOutlet weak var mapView: GMSMapView!
+    @IBOutlet weak var venueCollectionView: UICollectionView!
+    @IBOutlet weak var backButton: UIButton!
+    @IBOutlet weak var nextButton: UIButton!
+    private var isInit = false
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: I dont see you use it.

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 6]
```diff
@@ -40,6 +44,19 @@ final class LocationViewController: UIViewController {
         super.viewDidLoad()
         LocationManager.shared.delegate = self
         configLocation()
+        configCollectionView()
+    }
+
+    private func configCollectionView() {
+        venueCollectionView.backgroundColor = UIColor.clear
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: Omit `UIColor`. It is not necessary.

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 7]
```diff
@@ -113,3 +131,30 @@ extension LocationViewController: LocationManagerDelegate {
         }
     }
 }
+extension LocationViewController: UICollectionViewDataSource {
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: The default value is always 1.

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 8]
```diff
@@ -113,3 +131,30 @@ extension LocationViewController: LocationManagerDelegate {
         }
     }
 }
+extension LocationViewController: UICollectionViewDataSource {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: Make a new line between functions. 
Add a `//: - Mark` before a `extension`

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 9]
```diff
@@ -113,3 +131,30 @@ extension LocationViewController: LocationManagerDelegate {
...
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeue(CollectionViewCell.self, forIndexPath: indexPath)
+        cell.viewModel = viewModel.viewModelForItem(at: indexPath)
+        return cell
+    }
+}
+extension LocationViewController: UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return venueCollectionView.frame.size
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: I think you should be set item size in configCollectionView().

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 10]
```diff
@@ -13,6 +13,10 @@ import RealmS
 final class LocationViewController: UIViewController {
     // MARK: - IBOutlets
     @IBOutlet weak var mapView: GMSMapView!
+    @IBOutlet weak var venueCollectionView: UICollectionView!
+    @IBOutlet weak var backButton: UIButton!
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: Should be renamed `previousButton`.

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: ATFoody/View/Controller/Home/CollectionViewCell/VenueCollectionViewCell.swift [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    @IBOutlet weak var addressLabel: UILabel!
+    @IBOutlet weak var ratingLabel: UILabel!
+
+    var viewModel = VenueCollectionViewModel() {
+        didSet {
+            nameLabel.text = viewModel.name
+            addressLabel.text = viewModel.address
+            ratingLabel.text = viewModel.rating
+            ratingLabel.backgroundColor = UIColor.hexToColor(hexString: viewModel.ratingColor)
+            thumbnailImageView.setImage(urlString: viewModel.thumnailPath, placeholder: nil) { [weak self] (image) in
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/CollectionViewCell/VenueCollectionViewCell.swift]: No need handle `completion`

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 12]
```diff
@@ -113,3 +132,30 @@ extension LocationViewController: LocationManagerDelegate {
...
+        cell.viewModel = viewModel.viewModelForItem(at: indexPath)
+        return cell
+    }
+}
+
+// MARK: UICollectionViewDelegate, UICollectionViewDelegateFlowLayout
+extension LocationViewController: UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return venueCollectionView.frame.size
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: If the size for each item is fixed. Please calculate item size just one time.

**Pull request merged**